config/config.go

```diff
@@ -0,0 +1,229 @@
+   1. The configuration file is a env file
+   2. The configuration file, once it is made available to the application,
+   will stay immutable during the uptime of the application (exception is test)
+*/
+
+import (
+	"os"
+	"testing"
+    "time"
+
+	"github.com/spf13/viper"
+)
+
+// Global variable made available to any other package importing this package
+var envVars viper.Viper
+const testmode bool = false
+
+// Implementing a state machine tracking how things are going in this package
+const (
+	configgood    uint8 = 0
+	configbad     uint8 = 1
+	noconfigfound uint8 = 2
+)
+// State is a public variable for diagnostic purposes.
+var State uint8 = configgood // if config found and loaded, value not changed
+
+/*
+   This is the private helper function that sets up viper. This function is
+   called by the "Start" function below once, and it should be inlined by the
+   compiler, so no expense of an extra function call:
+   https://medium.com/@felipedutratine/does-golang-inline-functions-b41ee2d743fa
+*/
+func setup(dir string) *viper.Viper {
+
+	/*  TIP:
+	    The viper package allocates the Viper struct as "var v" during the import
+	    of the package. The line below is actually a function that calls a method
+	    using "var v".
+
+	    viper.SetConfigFile("yaml")
+
+	    However, we want the Viper struct to be unreachable after the
+	    values from the yaml file is retrieved, so the GC can clean it up after.
+	*/
+
+	// Print statement for diagnostic purposes only... to see the viper struct
+	// instantiated once. Comment out for production.
+	//println("Creating the viper struct!")
+
+	// Viper setup
+	var v = viper.New()
+	v.SetConfigName("local")
+	v.SetConfigType("env")
+	v.AddConfigPath(dir)
+	// Viper is lazy, do the read and parse of the env now
+	var err = v.ReadInConfig()
+
+	// If viper cannot read the configuration file...
+	if err != nil {
+		State = configbad
+	}
+
+	return v
+
+}
+
+/*
+   init:
+   When the packages is imported, try a couple of locations to find the local.env
+   configuration file. If the configuration file is not found, which is the case
+   with PROD, it will default the previous behavior of calling os.Getenv.
+*/
+func init() {
+
+	// Possible config locations. If there are more places to look, add here:
+	var locationSlice = []string{
+        // TEST DEV Location
+        "/go/src/github.com/CMSgov/bcda-app/shared_files/decrypted",
+        // This will be the location on PROD, which is currently not set
+        // So for now, it will wait 30 seconds and just default to os calls
+		".",
+	}
+
+    // TEST MODE
+    if testmode {
+        State = configbad
+        return
+    }
+
+	// Iterate through the possible locations
+	for i, v := range locationSlice {
+
+		// If the file exists
+		if success := fileexistattempt(3, v, i); success {
+			envVars = *setup(v)
+			break
+		}
+
+		// Exhausted the for loop, config file not found :/
+		if i == len(locationSlice) - 1 {
+			State = noconfigfound
```

Comment:
You can move this assignment outside of the loop, that way by default you're setting the noconfigfound and don't have to check if you've exhausted all of your options.